### PR TITLE
Move strings exclusively used by ride meta data to RideStringIds.h

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This PR introduces `RideStringIds.h` as the file containing the string id definitions for rides, at least until ride objects are introduced. This prevents having to recompile the `ride/` directory every time `StringIds.h` changes.

File was generated using Python to prevent any errors as a result of manual copy pasting.

### CODE REVIEW

File: src/openrct2/localisation/StringIds.h [hunk 1]
```diff
@@ -18,196 +18,15 @@ enum : uint16_t
...
-    STR_RIDE_NAME_CROOKED_HOUSE = 73,
-    STR_RIDE_NAME_MONORAIL_CYCLES = 74,
-    STR_RIDE_NAME_COMPACT_INVERTED_COASTER = 75,
-    STR_RIDE_NAME_WATER_COASTER = 76,
-    STR_RIDE_NAME_AIR_POWERED_VERTICAL_COASTER = 77,
-    STR_RIDE_NAME_INVERTED_HAIRPIN_COASTER = 78,
-    STR_RIDE_NAME_MAGIC_CARPET = 79,
-    STR_RIDE_NAME_SUBMARINE_RIDE = 80,
-    STR_RIDE_NAME_RIVER_RAFTS = 81,
     STR_RIDE_NAME_50 = 82,
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply src/openrct2/localisation/StringIds.h:24]: They're still used by the Ride window:
```cpp
        static constexpr StringId GetRideTypeNameForDropdown(ride_type_t rideType)
        {
            switch (rideType)
            {
                case RIDE_TYPE_1D:
                    return STR_RIDE_NAME_1D;
                case RIDE_TYPE_1F:
                    return STR_RIDE_NAME_1F;
                case RIDE_TYPE_22:
                    return STR_RIDE_NAME_22;
                case RIDE_TYPE_50:
                    return STR_RIDE_NAME_50;
                case RIDE_TYPE_52:
                    return STR_RIDE_NAME_52;
                case RIDE_TYPE_53:
                    return STR_RIDE_NAME_53;
                case RIDE_TYPE_54:
                    return STR_RIDE_NAME_54;
                case RIDE_TYPE_55:
                    return STR_RIDE_NAME_55;
                case RIDE_TYPE_59:
                    return STR_RIDE_NAME_59;
                default:
                    return GetRideTypeDescriptor(rideType).Naming.Name;
            }
        }
```

File: src/openrct2/ride/RideStringIds.h [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ *****************************************************************************/
+
+#pragma once
+
+#include "../common.h"
+
+enum : StringId
+{
+    // Coasters
+    STR_RIDE_DESCRIPTION_AIR_POWERED_VERTICAL_COASTER = 587,
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideStringIds.h:17]: It seems a bit odd to sort the description before the name, can you reverse that?

File: src/openrct2/ride/RideStringIds.h [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    STR_RIDE_DESCRIPTION_STEEPLECHASE = 522,
+    STR_RIDE_DESCRIPTION_SUSPENDED_SWINGING_COASTER = 514,
+    STR_RIDE_DESCRIPTION_TWISTER_ROLLER_COASTER = 6116,
+    STR_RIDE_DESCRIPTION_VERTICAL_DROP_ROLLER_COASTER = 556,
+    STR_RIDE_DESCRIPTION_VIRGINIA_REEL = 571,
+    STR_RIDE_DESCRIPTION_WATER_COASTER = 586,
+    STR_RIDE_DESCRIPTION_WILD_MOUSE = 566,
+    STR_RIDE_DESCRIPTION_WOODEN_ROLLER_COASTER = 564,
+    STR_RIDE_DESCRIPTION_WOODEN_WILD_MOUSE = 521,
+    STR_SPINNING_WILD_MOUSE_GROUP = 6162,
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideStringIds.h:102]: This name are a leftover from Ride Groups. Seems like a good moment to change to `STR_RIDE_NAME_SPINNING_WILD_MOUSE` and `STR_RIDE_DESCRIPTION_SPINNING_WILD_MOUSE`.

File: src/openrct2/ride/RideStringIds.h [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    STR_RIDE_DESCRIPTION_FERRIS_WHEEL = 549,
+    STR_RIDE_DESCRIPTION_FLYING_SAUCERS = 582,
+    STR_RIDE_DESCRIPTION_GHOST_TRAIN = 562,
+    STR_RIDE_DESCRIPTION_HAUNTED_HOUSE = 559,
+    STR_RIDE_DESCRIPTION_MAZE = 532,
+    STR_RIDE_DESCRIPTION_MERRY_GO_ROUND = 545,
+    STR_RIDE_DESCRIPTION_MINI_GOLF = 579,
+    STR_RIDE_DESCRIPTION_MINI_HELICOPTERS = 573,
+    STR_RIDE_DESCRIPTION_MONORAIL_CYCLES = 584,
+    STR_RIDE_DESCRIPTION_MONSTER_TRUCKS_GROUP = 6115,
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideStringIds.h:136]: Drop the _GROUP postfix (see previous comment as to why)

**Review by Gymnasiast — APPROVED** [review]
Right, that’s enough bikeshedding, looks good to me!

**Pull request merged**